### PR TITLE
Fix build by replacing ACName with ACIdentifier as in the spec

### PR DESCRIPTION
**Pull request by slallema**
Hi,
docker2aci wasn't building anymore so I tried to fix this by changing ACName to ACIdentifier where needed.
I think this was broken since https://github.com/appc/spec/pull/398 has been merged.

### CODE REVIEW

**Comment by jonboulle** [comment]
thanks! /cc @yifan-gu @iaguis

File: lib/common/common.go [hunk 1]
```diff
@@ -262,7 +262,7 @@ func convertVolumesToMPs(dockerVolumes map[string]struct{}) ([]appctypes.MountPo
 
 	for p := range dockerVolumes {
 		n := filepath.Join("volume-", p)
-		sn, err := appctypes.SanitizeACName(n)
+		sn, err := appctypes.SanitizeACIdentifier(n)
```
Comment (yifan-gu) [inline lib/common/common.go]: I think Volume still uses ACName

**Comment by yifan-gu** [comment]
@slallema Thanks for the fix! LGTM except for one comment above :)

File: lib/docker2aci.go [hunk 2]
```diff
@@ -281,7 +281,7 @@ func mergeManifests(manifests []schema.ImageManifest) schema.ImageManifest {
 	}
 
 	// this can't fail because the old name is legal
-	nameWithoutLayerID, _ := appctypes.NewACName(stripLayerID(manifest.Name.String()))
+	nameWithoutLayerID, _ := appctypes.NewACIdentifier(stripLayerID(manifest.Name.String()))
```
Comment (iaguis) [inline lib/docker2aci.go]: You can use MustACIdentifier and remove the comment above

**Comment by iaguis** [comment]
Apart from the two comments LGTM

**Comment by slallema** [comment]
Using ACName for volume name isn't working because right now the volumes names include the path : ```name: "volume-/data", path: "/data", readOnly: false``` and the spec regex for ACName has changed recently. 
Any idea on how we should handle that ?

**Comment by iaguis** [comment]
So the way the PR is not it works except for Comments (there's the string `docker/comment`), also the ports conversion uses things like `80/tcp`. I propose to replace slashes with dashes, `docker-comment` and `80-tcp`.

Volume names get converted with `SanitizeACName` so they're fine, every non-valid char would be converted to a dash, the only think that needs to change is

```
-               n := filepath.Join("volume-", p)
+               n := filepath.Join("volume", p)
```

so resulting volumes would be `volume-var-log-journal` for example.

**Comment by slallema** [comment]
I've done the changes, and it's working again for volumes and ports ! :) 
now the output looks like : 
```
Converted volumes:
	name: "volume-data", path: "/data", readOnly: false

Converted ports:
	name: "6379-tcp", protocol: "tcp", port: 6379, count: 1, socketActivated: false

Generated ACI(s):
redis-latest.aci
```

**Comment by jonboulle** [comment]
@iaguis happy with this?

**Comment by iaguis** [comment]
LGTM

**Pull request merged**